### ISSUE TITLE
Evetbus processor registration problem with socket

### ISSUE THREAD

**Issue by vertxjava** [eclipse-vertx/vert.x#2731]
Hi：

Please look at the chart below.

Using localConsumer here will make other vertx instances in the cluster invisible.Why not consumer but localConsumer? 

![image](https://user-images.githubusercontent.com/26713596/48928256-18da1880-ef18-11e8-80dc-bdf111c37546.png)

**Comment by tsegismont** [comment]
This does not look like Vert.x core code. Which project are you talking about? Can you elaborate on your issue? Expectations vs observations?

**Comment by vertxjava** [comment]
Thank you for your reply.
This code is in line 83 of the NetSocketImpl class, and the package is io.vertx.core.net.impl.
I don't think this should be a local Consumer.Because other vertx instances in the cluster cannot send eventbus messages.

![image](https://user-images.githubusercontent.com/26713596/48932949-4e432e00-ef39-11e8-9c2c-8e37bc37fef4.png)


I think this should be:
`vertx.eventBus().<Buffer>consumer(writeHandlerID).handler(writeHandler)`

But now we use localConsumer.

Thanks!

**Comment by vertxjava** [comment]
![image](https://user-images.githubusercontent.com/26713596/48933321-c6f6ba00-ef3a-11e8-9b3a-86d9bb81f24c.png)

Every socket automatically registers a handler on the event bus, 

But using the localConsumer method, if it is another vert. x instance in the cluster, it will not be able to send messages to writeHandlerID. Why is this a local Consumer instead of a consumer?

**Comment by vietj** [comment]
the purpose is to have this feature to be local only

**Comment by vertxjava** [comment]
Why not use consumer?

**Comment by vietj** [comment]
because it's potentially distributed

**Comment by okou19900722** [comment]
so the doc was wrong?

**Comment by vertxjava** [comment]
> This enables you to write data to a socket which is potentially in a completely different verticle or even in a different Vert.x instance by sending the buffer to the address of that handler.

The documentation says it can be used in different vert.x instances, but not now. Is the document wrong or is there a problem with the code?

**Comment by vietj** [comment]
I agree with you @vertxjava that's what is written, TBH I don't know what changed here and we should do git history research to understand the reason of this change

**Comment by tsegismont** [comment]
@vietj could be due to the pressure involved on the cluster manager in case the servers receives a great number of connections? Anyway, I wasn't aware of this feature and tbh it doesn't look reasonable to me.

**Comment by vietj** [comment]
of this feature cluster wise or local wise ?

**Comment by vertxjava** [comment]
Cluster.Because I can send messages in other vertx instances in the cluster, this is very convenient.

**Comment by tsegismont** [comment]
> of this feature cluster wise or local wise ?

@vietj both

**Comment by wang007** [comment]
I think it's up to the user to decide, NetSocketImpl don't need consumer, Implemented externally by the user via EventBus.

**Comment by wang007** [comment]
If cluster is implemented, netsocketImpl will be very heavy. Most users don't need this feature in NetSocketImpl
